lib/discourse_assign/entry_point.rb

```diff
@@ -2,15 +2,15 @@
 
 module DiscourseAssign
   class EntryPoint
+    # TODO: These four plugin api usages should ideally be in the assign plugin, not the solved plugin.
+    # They have been moved here from plugin.rb as part of the custom fields migration.
+
     def self.inject(plugin)
       plugin.register_modifier(:assigns_reminder_assigned_topics_query) do |query|
         next query if !SiteSetting.ignore_solved_topics_in_assigned_reminder
-        query.where.not(
-          id:
-            TopicCustomField.where(
-              name: ::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD,
-            ).pluck(:topic_id),
-        )
+        # TODO: this line was modified for the custom fields migration,
+        # but returning this huge array is not good at all.
+        query.where.not(id: DiscourseSolved::SolvedTopic.pluck(:topic_id))
```

Comment:
```suggestion
        query.where.not(id: DiscourseSolved::SolvedTopic.select(:topic_id))
```

There is no need for us to retrieve the values from the database just to pass it back to form a query.